docs/DeveloperGuide.adoc

```diff
@@ -262,6 +257,72 @@ We are using `java.util.logging` package for logging. The `LogsCenter` class is
 
 Certain properties of the application can be controlled (e.g user prefs file location, logging level) through the configuration file (default: `config.json`).
 
+=== UI
+The GUI of the application is started up by the `UiManager` class, when the application is initialised.
+The `GUI` class handles operations on the GUI, such as rendering nodes to the application window,
+and passing user command input to the `Responder` class.
+
+=== Timed/Untimed Test
+==== Proposed Implementation
+
+The timed/untimed test is a exam mechanism facilitated by the `ExamRunner` class which extends from the `Exam` interface.
+
+Given below is an example usage scenario and how the timed/untimed mechanism behaves at each step.
+
+Step 1. User keys in the command `test <duration> <deck name>` (Eg. test 120 German)
+
+Step 2. If `<duration>` is 0, untimed test will be initiated. Else, a timed test will be initiated.
+
+Step 3. User will be showed the card front, and prompted for a String answer.
+
+Step 4. Answer will be evaluated against the card back, and if it is correct, score will be updated.
+
+The following activity diagram displays the action flow of the test mechanism.
+
+image::ExamRunnerActivityDiagram.png[]
```

Comment:
![image](https://user-images.githubusercontent.com/8232172/67543618-e8e95580-f724-11e9-8516-fb9bf92a186f.png)

I think the diagram might be missing some syntax. What is the condition for updating the score? Also, it might be simpler to use "else" instead of "no cards" and "no cards or no time".